app/assets/javascripts/modules/master_overlay.coffee

```diff
@@ -24,12 +25,18 @@ do (document, window, $=jQuery, config=@edsc.config, plugin=@edsc.util.plugin, p
       @contentHeightChanged()
 
     maximize: ->
+      @_hideFlyouts()
       @_minimized = false
       @_updateMinMaxState()
       @contentHeightChanged()
 
     showParent: -> @toggleParent(true)
 
+    _hideFlyouts: ->
```

Comment:
The CollectionsCollapsedModel has a hideFlyout method. You should use that. This code ends up making assumptions about the internals of a component.

The cleanest way to handle this would be to have some sort of callback/event mechanism that gets triggered when the current pane is hidden and one that gets triggered when it's minimized or maximized. That would decouple the overlay from the specific components.

As I think you're encountering below, hiding flyouts this way leaks memory.